openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -182,6 +199,12 @@ def update_account_settings(requesting_user, update, username=None):
         for serializer in user_serializer, legacy_profile_serializer:
             serializer.save()
 
+        if "account_privacy" in update:
+            if account_privacy is not None:
+                account_privacy_serializer.save()
+            else:
+                delete_user_preference(requesting_user, "account_privacy")
+
```

Comment:
This isn't an issue since the call to `update_account_settings` is wrapped with a call to `transaction.commit_on_success()`.  See https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/user_api/accounts/views.py#L164

So as long as any exception is raised (validation or otherwise), the entire transaction is rolled back and the data is not saved.